### PR TITLE
Eml2 support

### PR DESCRIPTION
**Pull request by jeanetteclark**
I think we can go ahead and merge this. This addresses some of the items in issue #141. 

Relevant changes: 
- allow `publish_update` to handle EML 2.2 documents
- require an argument for `format_eml()` to easily set EML version when using `publish_object`

### CODE REVIEW

File: R/editing.R [hunk 1]
```diff
@@ -190,7 +190,7 @@ update_object <- function(mn, pid, path, format_id = NULL, new_pid = NULL, sid =
   sysmeta <- clear_replication_policy(sysmeta)
 
   # Add packageId to metadata if the object is an xml file
-  if (grepl("^eml:\\/\\/ecoinformatics.org\\/eml", format_id)) {
+  if (grepl("^eml:\\/\\/ecoinformatics.org\\/eml|^https://eml.ecoinformatics.org/eml-2.2.0", format_id)) {
```
Comment (amoeba) [inline R/editing.R]: we might future-proof this. Say, if we release EML 2.2.1, we'll need to update this. I think the second part of this pattern could safely be changed to `^https://eml.ecoinformatics.org` to make it work for the future.

**Comment by amoeba** [comment]
Looks good. @jeanetteclark did you want this to also update the `create_dummy_` helpers? We can do that after merge NP.

File: R/editing.R (hunk 1 above)
Comment (jeanetteclark) [reply R/editing.R]: good idea

**Comment by jeanetteclark** [comment]
lets update the dummy helpers after merge

**Pull request merged**